dandischema/models.py

```diff
@@ -1191,17 +1197,25 @@ class PublishedAsset(Asset, Publishable):
     schemaKey = "Asset"
 
     @validator("digest")
-    def digest_bothhashes(cls, values):
-        try:
-            digest = values[DigestType.dandi_etag]
-            if "-" not in digest or len(digest.split("-")[0]) != 32:
+    def digest_allhashes(cls, values):
+        digest = values.get(DigestType.dandi_etag, None)
+        if digest is not None:
+            if not re.fullmatch(DandiETag.REGEX, digest):
                 raise ValueError("Digest is missing dandi-etag value")
```

Comment:
message seems to be incorrect, 

```suggestion
                raise ValueError(
                    f"Digest must have an appropriate dandi-etag value. "
                    f"Got {digest}")
```
possibly again asking for some `_check_dandi_etag(values)` to be reused?